views/components/MovingCamera.jsx

```diff
@@ -1,33 +1,45 @@
-import React, { PropTypes } from 'react'
-var ReactTHREE = require('react-three')
-import { cameraRotation } from '../actions/Camera'
-var PerspectiveCamera = ReactTHREE.PerspectiveCamera;
-
-let dispatcher;
-document.onkeypress = (oPEvt)=> { //shouldnt this be on the container? =/
-    if(!dispatcher)
-        return;
-    var oEvent = oPEvt || window.event, charCode = oEvent.charCode;
-    if (charCode == 105 || charCode == 106 || charCode == 107 || charCode == 108 || charCode == 13 )
-        dispatcher(cameraRotation(charCode))
-};
+import React from 'react';
+import { cameraRotation } from '../actions/Camera';
+import * as BrowserActions from '../actions/Browser';
+
+const ReactTHREE = require('react-three');
+
+const PerspectiveCamera = ReactTHREE.PerspectiveCamera;
+
+class MovingCamera extends React.Component {
+  componentDidMount = () => {
+    this.props.dispatch(
+      BrowserActions.addEventListener('MovingCamera', 'keypress', this.rotateCamera)
+    );
+  };
 
+  componentWillUnmount = () => {
+    this.props.dispatch(
+      BrowserActions.removeEventListener('MovingCamera', 'keypress', this.rotateCamera)
+    );
+  };
 
-const MovingCamera = (config,dispatch)=>{
-    dispatcher = dispatch;
-    return <PerspectiveCamera name="maincamera" {...config} />
+  rotateCamera = (event) => {
+    this.props.dispatch(cameraRotation(event.keyCode));
```

Comment:
Having a controller for all the key events might be a  great idea since there will be more and more components needing to listen to this events. Also a better way to listen for events would be using [keydrown](https://github.com/jeremyckahn/keydrown) that provides a more reliable event timing an solves the browser compatibility issue. But this is out of scope for this PR.